core/src/main/java/hudson/tasks/LogRotator.java

```diff
@@ -76,10 +77,25 @@
      */
     private final Integer artifactNumToKeep;
 
+    /**
+     * If not -1 nor null, build logs are only kept up to this days.
+     * Null handling is necessary to remain data compatible with old versions.
+     * @since FIXME
+     */
+    private final Integer logDaysToKeep;
+
+    /**
+     * If not -1, only this number of build logs are kept.
+     * Null handling is necessary to remain data compatible with old versions.
+     * @since FIXME
+     */
+    private final Integer logNumToKeep;
+
     @DataBoundConstructor
-    public LogRotator (String daysToKeepStr, String numToKeepStr, String artifactDaysToKeepStr, String artifactNumToKeepStr) {
+    public LogRotator (String daysToKeepStr, String numToKeepStr, String artifactDaysToKeepStr, String artifactNumToKeepStr, String logDaysToKeepStr, String logNumToKeepStr) {
```

Comment:
🐛 this method is a part of public API, so this change breaks binary compatibility. The original method should be retained OR you could use `@DataBoundSetter`s